xbmc/Application.cpp

```diff
@@ -2582,17 +2582,8 @@ void CApplication::Stop(int exitCode)
 
 bool CApplication::PlayMedia(CFileItem& item, const std::string &player, int iPlaylist)
 {
-  // If item is a plugin, expand out
-  for (int i=0; URIUtils::IsPlugin(item.GetDynPath()) && i<5; ++i)
-  {
-    bool resume = item.m_lStartOffset == STARTOFFSET_RESUME;
-
-    if (!XFILE::CPluginDirectory::GetPluginResult(item.GetDynPath(), item, resume) ||
-        item.GetDynPath() == item.GetPath()) // GetPluginResult resolved to an empty path
-      return false;
-  }
-  // if after the 5 resolution attempts the item is still a plugin just return, it isn't playable
-  if (URIUtils::IsPlugin(item.GetDynPath()))
+  // if the item is a plugin we need to resolve the plugin paths
+  if (!XFILE::CPluginDirectory::GetResolvedPluginResult(item))
```

Comment:
@CastagnaIT sorry I missed this earlier and also for so many requests. I don't like the fact we are querying the plugin directory by default for each item that is requested for playback at a time we don't yet know the type of item. We should only enter this method if we know for sure at least one of the paths in the item is a `plugin://`.

Can you please add a new method to `URIUtils` called `HasPlugin` or `HasPluginPath`? This would return true if either `path` or `dynPath` is `plugin://`. Please then branch the `GetResolvedPluginResult` call only if `HasPluginPath` is true.
Thanks

Reply:
if i add `HasPluginPath` inside of `URIUtils ` will cause again jenkins error for the CFileItem reference
can i put the method in FileItem file?

Reply:
You could use a forward declarion for CFileItem on URIUtils.h, or just define the method in a way that it receives both paths directly

Reply:
Gave it a go, just forward declare CFIleItem on URIUtils.h, in URIUtils.cpp CFileItem is already included:

URIUtils.h:
```cpp
diff --git a/xbmc/utils/URIUtils.h b/xbmc/utils/URIUtils.h
index 8432c949e6..7e774143f3 100644
--- a/xbmc/utils/URIUtils.h
+++ b/xbmc/utils/URIUtils.h
@@ -13,6 +13,7 @@
 
 class CURL;
 class CAdvancedSettings;
+class CFileItem;
 
 class URIUtils
 {
@@ -28,6 +29,9 @@ public:
   static std::string GetExtension(const CURL& url);
   static std::string GetExtension(const std::string& strFileName);
 
+  // please add docs
+  const static bool HasPluginPath(const CFileItem& item);
```

```cpp
--- a/xbmc/utils/URIUtils.cpp
+++ b/xbmc/utils/URIUtils.cpp
@@ -66,6 +66,11 @@ std::string URIUtils::GetExtension(const std::string& strFileName)
   return strFileName.substr(period);
 }
 
+const bool URIUtils::HasPluginPath(const CFileItem& item)
+{
+  return IsPlugin(item.GetPath()) || IsPlugin(item.GetDynPath());
+}
+
```